jwst/spectral_leak/spectral_leak_step.py

```diff
@@ -0,0 +1,100 @@
+#! /usr/bin/env python
+
+from stdatamodels.jwst import datamodels
+from jwst.datamodels import ModelContainer
+from ..stpipe import Step
+import numpy as np
+
+from . import spectral_leak
+
+__all__ = ["SpectralLeakStep"]
+
+
+class SpectralLeakStep(Step):
+    """
+    The MIRI MRS has a spectral leak in which 6 micron light leaks into the
+    12 micron channel.  This step applies a correction to the 12 micron channel.
+    """
+
+    class_alias = "spectral_leak"
+
+    reference_file_types = ['mrsptcorr']
+
+    def process(self, input):
+        """Execute the step.
+
+        Parameters
+        ----------
+        input : model container
+            List of input 1D spectral data models
+
+        Returns
+        -------
+        result : data model
+            The corrected data model. This will be "input" if ...
+            This will be `input` if the step was skipped; otherwise,
+            it will be a corrected 1-D extracted spectra that contains
+            the  3B MRS range.
```

Comment:
```suggestion
            The corrected data model. This will be "input" if the step is skipped,
            otherwise it will be a corrected 1D extracted spectrum that contains
            the MRS channel 3B range.
```